fiftyone/core/odm/embedded_document.py

```diff
@@ -40,11 +45,22 @@ class DynamicEmbeddedDocument(
     database.
 
     Dynamic documents can have arbitrary fields added to them.
+
+    Raises FiftyoneDocumentException if a kwarg attribute is reserved.
     """
 
     meta = {"abstract": True, "allow_inheritance": True}
 
     def __init__(self, *args, **kwargs):
+        for key, value in kwargs.items():
```

Comment:
Have you benchmarked this implementation? While validation is nice, in the context of something I/O intensive like iterating over a dataset or adding a bunch of samples to a dataset, this would introduce millions of extra iterations, string comparisons, etc.

Doing validation to prevent bad data from being added to the database is a necessary expense. But in this case, the damage is already done. We're raising a more informative error message, but the user still can't work with the dataset (in the case where we're iterating over an existing dataset).

A faster alternative if we really want this validation is to [use try-except](https://docs.python.org/2/faq/design.html#how-fast-are-exceptions).

Reply:
Ah, the performance consideration makes sense, I haven't done the benchmarking but I think it's safer to use `try-except`. I did think about it but I wanted to do more precise validation by iterating through the key-value pairs.